### PR TITLE
Feature/particle implementation

### PR DESCRIPTION
**Pull request by Gorshtak**

### CODE REVIEW

File: src/pages/AARelatedPages/Withdraw/Withdraw.tsx [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+        if (withdrawalWalletAddress != '' && ethers.utils.isAddress(withdrawalWalletAddress)) {
+            walletValidation = true;
+        }
+
+        if (amount > 0 && !(amount > paymentTokenBalance)) {
+            amountValidation = true;
+        }
+
+        setValidation({ walletAddress: walletValidation, amount: amountValidation });
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (lotinasa) [inline src/pages/AARelatedPages/Withdraw/Withdraw.tsx:101]: Can be removed

File: src/pages/AARelatedPages/Withdraw/components/WithdrawalConfirmationModal/WithdrawalConfirmationModal.tsx [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+    const networkName = useMemo(() => {
+        return getNetworkNameByNetworkId(network);
+    }, [network]);
+
+    const parsedAmount = useMemo(() => {
+        return coinParser('' + amount, network, token);
+    }, [amount, network, token]);
+
+    // useEffect(() => {
```
Comment (lotinasa) [inline src/pages/AARelatedPages/Withdraw/components/WithdrawalConfirmationModal/WithdrawalConfirmationModal.tsx:45]: Just temporary or forgotten?

File: src/pages/Profile/components/MyPortfolio/MyPortfolio.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import Balance from './component/Balance';
+
+const MyPortfolio: React.FC = () => {
+    return (
+        <>
```
Comment (lotinasa) [inline src/pages/Profile/components/MyPortfolio/MyPortfolio.tsx:6]: can be removed

File: src/pages/Root/Provider/ParticleProvider/ParticleProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+                clientKey: process.env.REACT_APP_CLIENT_KEY as string,
+                appId: process.env.REACT_APP_PARTICLE_APP_ID as string,
+                chainName: getNetworkNameByNetworkId(networkId, true)?.toLowerCase() || 'optimism',
+                chainId: networkId || 10,
+                wallet: {
+                    //optional: by default, the wallet entry is displayed in the bottom right corner of the webpage.
+                    displayWalletEntry: false, //show wallet entry when connect particle.
+                    uiMode: 'dark', //optional: light or dark, if not set, the default is the same as web auth.
+                    supportChains: [
+                        { id: 10, name: 'optimism' },
```
Comment (lotinasa) [inline src/pages/Root/Provider/ParticleProvider/ParticleProvider.tsx:30]: Network IDs can be replaced with enum values e.g. Network.OptimismMainnet

**Review by lotinasa — DISMISSED** [review]

File: src/components/ConnectWalletModal/ConnectWalletModal.tsx [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+            console.log('Error occurred');
+        }
+    };
+
+    useEffect(() => {
+        if (isSuccess) {
+            if (modalOrigin == 'sign-up') navigateTo(ROUTES.Wizard);
+            onClose();
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```
[the remark this answers is not in the document]
Comment (Gorshtak) [reply src/components/ConnectWalletModal/ConnectWalletModal.tsx:92]: Not sure, will put them in dependency array and remove the comment for disabling eslint

File: src/pages/AARelatedPages/Deposit/Deposit.tsx [hunk 6]
```diff
@@ -0,0 +1,400 @@
...
+    useEffect(() => {
+        if (totalBalanceValue == 0) {
+            setTotalValue(0);
+            return;
+        }
+        if (totalValue == 0 && totalBalanceValue && totalBalanceValue > 0) {
+            setTotalValue(totalBalanceValue);
+            setShowSuccessfulDepositModal(true);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```
[the remark this answers is not in the document]
Comment (Gorshtak) [reply src/pages/AARelatedPages/Deposit/Deposit.tsx:101]: Not sure, added it back and changed the logic a little bit, should be tested before merging

File: src/pages/AARelatedPages/Deposit/Deposit.tsx [hunk 7]
```diff
@@ -0,0 +1,400 @@
...
+    const [, setSelectedTokenFromQuery] = useQueryParam(
+        'coin-index',
+        selectedTokenFromUrl ? selectedTokenFromUrl : '0'
+    );
+
+    useEffect(() => {
+        if (selectedTokenFromUrl != selectedToken.toString()) {
+            setSelectedToken(Number(selectedTokenFromUrl));
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```
[the remark this answers is not in the document]
Comment (Gorshtak) [reply src/pages/AARelatedPages/Deposit/Deposit.tsx:61]: Not sure, but removed selectedToken from dependencies as this should only be checked for selectedTokenFromUrl

File: src/pages/AARelatedPages/Withdraw/Withdraw.tsx [hunk 8]
```diff
@@ -0,0 +1,210 @@
...
+    const [, setSelectedTokenFromQuery] = useQueryParam(
+        'coin-index',
+        selectedTokenFromUrl ? selectedTokenFromUrl : '0'
+    );
+
+    useEffect(() => {
+        if (selectedTokenFromUrl != selectedToken.toString()) {
+            setSelectedToken(Number(selectedTokenFromUrl));
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
```
[the remark this answers is not in the document]
Comment (Gorshtak) [reply src/pages/AARelatedPages/Withdraw/Withdraw.tsx:66]: Removed it from dependencies and from if statement

File: src/pages/AARelatedPages/Withdraw/components/WithdrawalConfirmationModal/WithdrawalConfirmationModal.tsx (hunk 2 above)
Comment (Gorshtak) [reply src/pages/AARelatedPages/Withdraw/components/WithdrawalConfirmationModal/WithdrawalConfirmationModal.tsx:45]: Will be needed for biconomy AA

**Review by vpoklopic — CHANGES_REQUESTED** [review]

**Review by vpoklopic — APPROVED** [review]

**Review by vpoklopic — DISMISSED** [review]

**Review by vpoklopic — APPROVED** [review]

**Pull request merged**